README.md

```diff
@@ -1,6 +1,53 @@
+ | strategy                                         | Control suppression scope that you need. If `startegy` is set to `newline`, it only accepts TreeWalker audit events for TreeWalker Check violations whose line number belong to added lines in patch file. `patchedline` will accept added/changed lines. if `strategy` is set to `context` , for checks listed in `supportContextStrategyChecks`, it will accept violations whose line number belong to added/changed/deleted lines and new code introduces violations outside of added/changed lines, but its child nodes in added/changed lines, for checks not listed in `supportContextStrategyChecks`, it will accept violations whose line number belong to added/changed/deleted lines in patch file. | [String](https://checkstyle.sourceforge.io/property_types.html#String) | newline       |
+ | supportContextStrategyChecks                     | String has user defined Checks that support context strategy | [String](https://checkstyle.sourceforge.io/property_types.html#String) | null          |
+ | neverSuppressedChecks                            | String has user defined TreeWalker Checks to never suppress if files are touched, split by comma. This property is useful for Checks that place violation on whole file or on not all (first/last) occurrence of cause/violated code. | [String](https://checkstyle.sourceforge.io/property_types.html#String) | null          |
+ | checkNamesForContextStrategyByTokenOrAncestorSet | String has user defined TreeWalker Checks that need modify violation nodes to their ancestor nodes to expand the context scope, split by comma | [String](https://checkstyle.sourceforge.io/property_types.html#String) | null          |
+
+#### Examples
+
+For example, the following configuration fragment directs the Checker to use a SuppressionPatchXpathFilter 
+with patch file config/defaultContext.patch and strategy is `newline`:
+```xml
+<module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchXpathFilter">
+    <property name="file" value="config/defaultContext.patch" />
+    <property name="strategy" value="newline" />
+</module>
+```
+the following configuration fragment direts the Checker to use a SuppressionPatchXpathFilter 
+with patch file config/defaultContext.patch, whose strategy is `context`,
+support context strategy check `MethodLength` and never suppress checks `EmptyBlock` and `HiddenField`:
+```xml
+<module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchXpathFilter">
+    <property name="file" value="${tp}/defaultContext.patch" />
+    <property name="strategy" value="context" />
+    <property name="supportContextStrategyChecks" value="MethodLength," />
+    <property name="neverSuppressedChecks" value="EmptyBlock,HiddenField" />
+</module>
+```
```

Comment:
Please add example for the `patchedline` strategy

Reply:
done